indico/modules/events/papers/controllers/api.py

```diff
@@ -106,3 +120,82 @@ def _check_paper_protection(self):
     def _process(self, files):
         create_paper_revision(self.paper, session.user, files)
         return '', 204
+
+
+def _parse_review_args(event, review_type):
+    args_schema = {
+        'proposed_action': EnumField(PaperAction, required=True),
+        'comment': fields.String(missing='')
+    }
+
+    for question in event.cfp.get_questions_for_review_type(review_type):
+        attrs = {}
+        if question.is_required:
+            attrs['required'] = True
+        else:
+            attrs['missing'] = None
+
+        if question.field_type == 'rating':
+            field_cls = fields.Integer
+        elif question.field_type == 'text':
+            validators = []
+            if question.field_data['max_length']:
+                validators.append(validate.Length(max=question.field_data['max_length']))
+            if question.field_data['max_words']:
+                validators.append(max_words(question.field_data['max_words']))
+
+            attrs['validate'] = validators
+            field_cls = fields.String
+        else:
```

Comment:
wouldn't it be safer to use an elif here as well and raise an exception in else or use else for the `text` type?

Reply:
Yeah, however it is pretty much improbable to happen :P